library(traits)
seqs<-ncbi_byid(ids = c("DQ683142.1","DQ683138.1","DQ683137.1","DQ683159.1","DQ683157.1"), verbose = TRUE)
print(paste(seqs$gene_desc,seqs$sequence))
dnaSet=DNAStringSet(seqs$sequence)
res=msa(dnaSet,method = "ClustalW")
print(res)
print(res,show="complete")
res2<-msaConvert(res,type = "seqinr::alignment")
msa(dnaSet)
d<-dist.alignment(res2,"identity")
as.matrix(d)[2:5,drop=FALSE]
hemoTree<-nj(d)
plot(hemoTree,main="Phylogenetic Tree of Koala retrovirus envelope glycoprotein")
orig_tiplabels<-c("1","2","3","4","5")
orig_tree<-rtree(n=5,tip.label=orig_tiplabels)
plot(orig_tree)
new_tiplabels<-c("Koala 6 22","Koala 1 216","Koala 1 21","Koala Jacob 32","Koala 1 35")
orig_tree$tip.label<-new_tiplabels
plot(orig_tree)  
library(phangorn)
Alignment_phyDat <- msaConvert(res, type="phangorn::phyDat")
write.phyDat(Alignment_phyDat, "alignment.fasta", format = "fasta")